src/Autodoc.zig

```diff
@@ -1520,15 +1520,39 @@ fn walkInstruction(
                 .expr = .{ .builtinIndex = bin_index },
             };
         },
+        .truncate => {
+            // in the ZIR this node is a builtin `bin` but we want send it as a `un` builtin
+            const pl_node = data[inst_index].pl_node;
+            const extra = file.zir.extraData(Zir.Inst.Bin, pl_node.payload_index);
 
+            var rhs: DocData.WalkResult = try self.walkRef(
+                file,
+                parent_scope,
+                parent_src,
+                extra.data.rhs,
+                false,
+            );
+
+            const bin_index = self.exprs.items.len;
+            try self.exprs.append(self.arena, .{ .builtin = .{ .param = 0 } });
+
+            const rhs_index = self.exprs.items.len;
+            try self.exprs.append(self.arena, rhs.expr);
+
+            self.exprs.items[bin_index] = .{ .builtin = .{ .name = @tagName(tags[inst_index]), .param = rhs_index } };
+
+            return DocData.WalkResult{
+                .typeRef = rhs.typeRef orelse .{ .type = @intFromEnum(Ref.type_type) },
```

Comment:
This is not correct, see the definition of what truncate does taken from the language reference:

```
@truncate(comptime T: type, integer: anytype) T
```
> This function truncates bits from an integer type, resulting in a smaller or same-sized integer type.

This is still the same, except that now you don't say explicitly the target type in the source code (but it's still available in ZIR, since it's still a bin instruction).

So in other words, the function performs a type change and the new type is the `lhs`, not the original type of the `rhs`.

Reply:
or should i say Left that adds up xD haha

Reply:
Im not sure if i need to add the lhs to self.exprs, but it seems to work